java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -120,6 +120,9 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch)
 
     long start = out.getCurrentPosition();
     int bodyLength = batch.computeBodyLength();
+    if (bodyLength % 8 != 0) {
+      bodyLength += 8 - (bodyLength % 8);
+    }
```

Comment:
In C++ we are going so far as to pad buffers to 64 bytes: https://github.com/apache/arrow/blob/master/cpp/src/arrow/ipc/writer.cc#L150. We might relax this to 8-byte padding in the interest of space savings, since 64-byte padding is not a hard requirement (per ARROW-1224)